### ISSUE TITLE
Use async/await in APIRoute key

### ISSUE THREAD

**Issue by tatimblin** [ryanto/next-s3-upload#16]
I was looking at #6 because I wanted to include a users uuid in the filepath for images they upload. This would require an asynchronous call to the database. I am having trouble setting this up, can you point me in the right direction to achieving this?

Also thanks for making this package! :)

```
import { APIRoute } from 'next-s3-upload';
import { v4 as uuidv4 } from 'uuid';
import { currentUser } from './db'

export default async function (req, filename) {
  const token = getCookie(req.headers.cookie, 'token'); <--- return the cookie named 'token'
  const user = await currentUser(token); <--- I want to await this

  return APIRoute.configure({
    key() {
      return `image-uploads/${user.ref.id}/${uuidv4()}-${filename.replace(/\s/g, '-')}`;
    }
  });
}
```

**Comment by ryanto** [comment]
Nice catch! I think this is bug in next-s3-upload because right now we only allow sync `key` functions.

Ideally we change key() to be async. I should be able to do this in the next week or so.

**Comment by ryanto** [comment]
This is released in version `0.1.6`.
